frontend/src/libs/packages/store/middlewares/notification-middleware.ts

```diff
@@ -0,0 +1,59 @@
+import { type TypedStartListening } from '@reduxjs/toolkit';
+import {
+  createListenerMiddleware,
+  isFulfilled,
+  isRejected,
+} from '@reduxjs/toolkit';
+
+import { type AsyncThunkConfig } from '~/libs/types/async-thunk-config.type';
+import { NotificationMessage } from '~/packages/notification/libs/enums/notification-message.enum';
+import { NotificationType } from '~/packages/notification/libs/enums/notification-type.enum';
+import { appActions } from '~/slices/app/app.js';
+
+const notificationMiddleware = createListenerMiddleware();
+
+type AppStartListening = TypedStartListening<
+  AsyncThunkConfig['state'],
+  AsyncThunkConfig['dispatch']
+>;
+
+const startAppListening =
+  notificationMiddleware.startListening as AppStartListening;
+
+startAppListening({
+  matcher: isRejected,
+  effect: (action, listenerApi) => {
+    const config = NotificationMessage[action.type];
```

Comment:
And what about success notifications ? We can't fire it for every action because user don't need it and it will create infinite loop considering that notification dispatch is also an action. So we need some sort of list of actions to react.  Or maybe it's just 2 separate tasks that shouldn't have been mixed ?

Reply:
We should automatically dispatch only notifications for errors - so in other cases it may just skip dispatching action and that's it

Reply:
Maybe we don't need the actual listening middleware, and just simplify it to actions interceptor (that will check if actions are rejected)

Something like this
```
const handleError = () => {
  return (next: typeof store.instance.dispatch) =>
    (action: AnyAction): unknown => {
      const result: unknown = next(action);

      if (isRejected(result) && !result.meta.rejectedWithValue) {
        const message = result.error.message ?? 'Try again later';

        return next(
          appActions.notify({ type: NotificationType.ERROR, message }),
        );
      }

      return result;
    };
};
```

Reply:
Okay, that's an option 👍 
For `success` and other cases - when we'd need it - we will dispatch notifications directly